integration/test/metrics_number_dimension/metrics_number_dimension_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metrics_number_dimension
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent/integration/test"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"log"
+	"strings"
+	"testing"
+	"time"
+)
+
+const configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+const configJSON = "/config.json"
+const namespace = "MetricNumberDimensionTest"
+const instanceId = "InstanceId"
+const append = "append"
+
+//Let the agent run for 2 minutes. This will give agent enough time to call server
+const agentRuntime = 2 * time.Minute
+
+var targetString = "max MaxDimensions %v is less than than number of dimensions %v thus only taking the max number"
+
+type input struct {
+	dataInput            string
+	findTarget           bool
+	inputNumberDimension int
+	metricName           string
+}
+
+type metric struct {
+	name  string
+	value string
+}
+
+//Must run this test with parallel 1 since this will fail if more than one test is running at the same time
+//This test uses a pem file created for the local stack endpoint to be able to connect via ssl
+func TestBundle(t *testing.T) {
+
+	parameters := []input{
+		{dataInput: "resources/10_dimension",
```

Comment:
I don't think that `dataInput` is a very clear name for what this represents - might want to revisit the cert tests too, but not a blocker. The value here is a directory/path to a specific config file, right?